scan-citrix-netscaler-version.py

```diff
@@ -325,9 +325,13 @@ def scan_netscaler_version(target: str, client: httpx.Client) -> NetScalerVersio
 
         # Temporarily enable certificate verification to extract some information that is not available otherwise
         with temporary_ssl_verify_mode(ssl_ctx, ssl.CERT_OPTIONAL):
-            cert = ssl_object.getpeercert()
-            if cert and "subjectAltName" in cert:
-                subject_alt_names = ", ".join(s[1] for s in cert["subjectAltName"])
+            try:
+                cert = ssl_object.getpeercert()
+            except AttributeError:
+                pass
+            else:
+                if cert and "subjectAltName" in cert:
+                    subject_alt_names = ", ".join(s[1] for s in cert["subjectAltName"])
```

Comment:
Oops, I forgot to handle non HTTPs cases, if we add a `ssl_object` guard it will work:

```suggestion
        if ssl_object:
            # Temporarily enable certificate verification to extract some information that is not available otherwise
            with temporary_ssl_verify_mode(ssl_ctx, ssl.CERT_OPTIONAL):
                cert = ssl_object.getpeercert()
                if cert and "subjectAltName" in cert:
                    subject_alt_names = ", ".join(s[1] for s in cert["subjectAltName"])
```

Reply:
That's exactly what I did, but using the EAFP strategy rather than LBYL. The rationale is that it is the most efficient strategy when you know that you will have a certificate most of the time and that the plain HTTP is the exception.